frontend/src/main/web/lib/stores/GlossaryStore.js

```diff
@@ -0,0 +1,522 @@
+  var url = Configs.baseUrl + "/glossary/upload",
+    uploadFile = data.uploadFile;
+
+  _state['uploadFile']['status'] = 0;
+
+  return new Promise(function(resolve, reject) {
+    Request.post(url)
+      .attach('file', uploadFile.file, uploadFile.file.name)
+      .field('fileName', uploadFile.file.name)
+      .field('srcLocale', data.srcLocale)
+      .field('transLocale', data.uploadFile.transLocale)
+      .set('Accept', 'application/json')
+      .end(function (err, res) {
+        _state['uploadFile']['status'] = -1;
+        _state['uploadFile'].show = false;
+        _state['uploadFile'].transLocale = null;
+        _state['uploadFile'].file = null;
+
+        if(err != null && err.error === true) {
+          console.error(url, err);
+        }
+        if(res != null) {
+          if (res.error) {
+            console.error(url, res.status, res.error.toString());
+            reject(Error(res.error.toString()));
+          } else {
+            resolve(res['body']);
+          }
+        }
+        resolve(null);
+      })
+      .on('progress', function(e) {
+        _state['uploadFile']['status'] = e.percent;
+      });
+  });
+}
+
+function processDelete(serverResponse) {
+  console.debug('Glossary entry deleted');
+}
+
+function processSaveOrUpdate(serverResponse) {
+  _state['newEntry'].isSaving = false;
+  _state['newEntry'].show = false;
+  _state['newEntry'].isAllowSave = false;
+  _state['newEntry'].pos = '';
+  _state['newEntry'].term = '';
+  _state['newEntry'].description = '';
+
+  console.debug('Glossary entry saved');
+}
+
+function processUpdate(serverResponse) {
+  console.debug('Glossary entry updated');
+}
+
+function initialise() {
+  //load permission here
+  _state['canAddNewEntry'] = canAddNewEntry();
+  _state['canUpdateEntry'] = canUpdateEntry();
+
+  loadLocalesStats()
+    .then(processLocalesStatistic)
+    .then(loadGlossaryByLocale)
+    .then(processGlossaryList)
+    .then(function (newState) {
+      GlossaryStore.emitChange();
+    });
+}
+
+var GlossaryStore = assign({}, EventEmitter.prototype, {
+  init: function() {
+    if (_state.locales === null) {
+      initialise();
+    }
+    return _state;
+  }.bind(this),
+
+  getEntry: function(resId) {
+    return _state['glossary'][resId];
+  },
+
+  getFocusedRow: function() {
+    return _state['focusedRow'];
+  },
+
+  getNewEntryState: function() {
+    return _state['newEntry'];
+  },
+
+  getUploadFileState: function() {
+    return _state['uploadFile'];
+  },
+
+  emitChange: function() {
+    this.emit(CHANGE_EVENT);
+  },
+
+  /**
+   * @param {function} callback
+   */
+  addChangeListener: function(callback) {
+    this.on(CHANGE_EVENT, callback);
+  },
+
+  /**
+   * @param {function} callback
+   */
+  removeChangeListener: function(callback) {
+    this.removeListener(CHANGE_EVENT, callback);
+  },
+
+  dispatchToken: Dispatcher.register(function(payload) {
+    var action = payload.action;
+    switch (action['actionType']) {
```

Comment:
This switch statement is way too long. Split the body of each case out to a function so it is easy to read the whole switch statement without loads of scrolling.